dependencies/clients/teku.go

```diff
@@ -0,0 +1,349 @@
+package clients
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+
+	"github.com/lukso-network/tools-lukso-cli/common/errors"
+	"github.com/lukso-network/tools-lukso-cli/common/system"
+	"github.com/lukso-network/tools-lukso-cli/common/utils"
+	"github.com/lukso-network/tools-lukso-cli/flags"
+	"github.com/lukso-network/tools-lukso-cli/pid"
+)
+
+const (
+	tekuDepsFolder = "teku" // folder in which both teku and JDK are stored
+	tekuFolder     = "teku" // folder in which teku is stored (in tekuDepsFolder)
+	jdkFolder      = "jdk"  // folder in which JDK is stored (in tekuDepsFolder)
+	jdkInstallURL  = "https://download.java.net/java/GA/jdk20.0.2/6e380f22cbe7469fa75fb448bd903d8e/9/GPL/openjdk-20.0.2_|OS|-|ARCH|_bin.tar.gz"
+)
+
+type TekuClient struct {
+	*clientBinary
+}
+
+func NewTekuClient() *TekuClient {
+	return &TekuClient{
+		&clientBinary{
+			name:           tekuDependencyName,
+			commandName:    "teku",
+			baseUrl:        "https://artifacts.consensys.net/public/teku/raw/names/teku.tar.gz/versions/|TAG|/teku-|TAG|.tar.gz",
+			githubLocation: tekuGithubLocation,
+		},
+	}
+}
+
+var Teku = NewTekuClient()
+
+var _ ClientBinaryDependency = &TekuClient{}
+
+func (t *TekuClient) PrepareStartFlags(ctx *cli.Context) (startFlags []string, err error) {
+	startFlags = t.ParseUserFlags(ctx)
+
+	startFlags = append(startFlags, fmt.Sprintf("--config-file=%s", ctx.String(flags.TekuConfigFileFlag)))
+
+	return
+}
+
+func (t *TekuClient) Install(url string, isUpdate bool) (err error) {
+	if utils.FileExists(t.FilePath()) && !isUpdate {
+		message := fmt.Sprintf("You already have the %s client installed, do you want to override your installation? [Y/n]: ", t.Name())
+		input := utils.RegisterInputWithMessage(message)
+		if !strings.EqualFold(input, "y") && input != "" {
+			log.Info("⏭️  Skipping installation...")
+
+			return nil
+		}
+	}
+
+	err = installAndUntarFromURL(url)
+	if err != nil {
+		return
+	}
+
+	switch isUpdate {
+	case true:
+		log.Infof("✅  %s updated!\n\n", t.Name())
+	case false:
+		log.Infof("✅  %s downloaded!\n\n", t.Name())
+	}
+
+	_, isInstalled := os.LookupEnv(system.JavaHomeEnv) // means that JDk is not set up
+	if !isInstalled {
+		message := "Teku is written in Java. This means that to use it you need to have:\n" +
+			"- JDK installed on your computer\n" +
+			"- JAVA_HOME environment variable set\n" +
+			"Do you want to install and set up JDK along with Teku? [Y/n]\n>"
```

Comment:
We need to mention about JAVA in `README`

Reply:
Done